pkg/kubelet/dockershim/convert.go

```diff
@@ -69,7 +69,7 @@ func toPullableImageID(id string, image *dockertypes.ImageInspect) string {
 	// Default to the image ID, but if RepoDigests is not empty, use
 	// the first digest instead.
 	imageID := DockerImageIDPrefix + id
-	if len(image.RepoDigests) > 0 {
+	if image != nil && len(image.RepoDigests) > 0 {
```

Comment:
Can you share more about why this nil check is necessary?

Reply:
We can call `toPullableImageID` to generate default imageID when `InspectImageByID` return `nil, err`